code/hud/hudartillery.cpp

```diff
@@ -69,26 +69,35 @@ void parse_ssm(const char *filename)
 		while(required_string_either("#end", "$SSM:")) {
 			required_string("$SSM:");
 			ssm_info s;
+			int string_index;
 
 			// name
 			stuff_string(s.name, F_NAME, NAME_LENGTH);
+			if (*s.name == 0) {
+				sprintf(s.name, "SSM " SIZE_T_ARG, Ssm_info.size());
+				Warning(LOCATION, "SSM entries must have a name!  Assigning \"%s\"\n", s.name);
```

Comment:
Exactly. They're optional. We don't usually issue warnings about optional data not being present (that would imply that it was actually required).

Now, granted, it's an exceedingly bad idea not to provide names for SSM strikes defined in a modular table, so issuing a warning when `Parsing_modular_table` might make sense (relying on either the automatically-assigned index or the autogenerated name for a modular entry should be strongly discouraged), but otherwise, we're adding a warning to working, technically-correct, already-released mods for no apparent reason.